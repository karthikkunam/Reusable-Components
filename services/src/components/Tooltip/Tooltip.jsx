import React, { useRef } from 'react';
import PropTypes from 'prop-types';
import OverlayTrigger from 'react-bootstrap/OverlayTrigger';
import Tooltip from 'react-bootstrap/Tooltip';

function CustomTooltip(props) {
  const target = useRef(null);
  const { children, placement, iconClassName, className, trigger, dynamicContent } = props;

  return (
    <div className={className}>
      <OverlayTrigger
        key={placement}
        placement={placement}
        trigger={trigger}
        overlay={
          <Tooltip id={`tooltip-${placement}`} className={`custom-tooltip bs-popover-${placement}`}>
            {children}
          </Tooltip>
        }>
        <button className="btn btn-sm btn-tooltip bs-tooltip" id="open-popover-link">
          <span>{dynamicContent}</span>
          <span ref={target} className={iconClassName}></span>
        </button>
      </OverlayTrigger>
    </div>
  );
}

CustomTooltip.displayName = 'Tooltip';
CustomTooltip.defaultProps = {
  placement: 'bottom',
  trigger: ['hover', 'click'],
};

CustomTooltip.propTypes = {
  children: PropTypes.node,
  placement: PropTypes.string,
  iconClassName: PropTypes.string,
  className: PropTypes.string,
  trigger: PropTypes.string,
  dynamicContent: PropTypes.string,
};

export default CustomTooltip;
