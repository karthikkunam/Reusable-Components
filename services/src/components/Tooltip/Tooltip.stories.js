import React from 'react';
import Tooltip from '.';

const wrapperStyle = {
  paddingLeft: '250px',
  paddingTop: '50px',
};

export default {
  title: 'Tooltip',
  component: Tooltip,
  decorators: [storyFn => <div style={wrapperStyle}>{storyFn()}</div>],
};

export const DefaultTooltip = () => (
  <>
    <div>Default Tooltip</div>
    <Tooltip iconClassName="icon icon-info">
      <div className="tooltip-content">
        <div className="header">Default Tooltip</div>
      </div>
    </Tooltip>
  </>
);
export const TooltipOnClick = () => (
  <>
    <div>Click on Icon</div>
    <Tooltip trigger='click' iconClassName="icon icon-info">
      <div className="tooltip-content">
        <div className="header">Default Tooltip</div>
      </div>
    </Tooltip>
  </>
);
export const DynamicContentTooltip = () => (
  <>
    <div>Click on Icon</div>
    <Tooltip trigger='click' dynamicContent="12" iconClassName="icon icon-info">
      <div className="tooltip-content">
        <div className="header">Default Tooltip</div>
      </div>
    </Tooltip>
  </>
);

export const TooltipWithPlacement = () => (
  <>
    <div>Top Tooltip</div>
    <Tooltip placement="top" iconClassName="icon icon-info">
      <div className="tooltip-content">
        <div className="header">Purchase Margin (%) =</div>
        <div className="body">[ (LDU * Item Retail – Item Cost + Allowance Amount) / (LDU * Item Retail) ] * 100</div>
      </div>
    </Tooltip>
    <div>Right Tooltip</div>
    <Tooltip placement="right" iconClassName="icon icon-info">
      <div className="tooltip-content">
        <div className="header">Purchase Margin (%) =</div>
        <div className="body">[ (LDU * Item Retail – Item Cost + Allowance Amount) / (LDU * Item Retail) ] * 100</div>
      </div>
    </Tooltip>
    <div>Left Tooltip</div>
    <Tooltip placement="left" iconClassName="icon icon-info">
      <div className="tooltip-content">
        <div className="header">Purchase Margin (%) =</div>
        <div className="body">[ (LDU * Item Retail – Item Cost + Allowance Amount) / (LDU * Item Retail) ] * 100</div>
      </div>
    </Tooltip>
    <div>Bottom Tooltip</div>
    <Tooltip placement="bottom" iconClassName="icon icon-info">
      <div className="tooltip-content">
        <div className="header">Purchase Margin (%) =</div>
        <div className="body">[ (LDU * Item Retail – Item Cost + Allowance Amount) / (LDU * Item Retail) ] * 100</div>
      </div>
    </Tooltip>
  </>
);
