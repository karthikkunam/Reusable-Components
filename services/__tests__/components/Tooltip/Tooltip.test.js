import React from 'react';
import { shallow } from 'enzyme';
import Tooltip from '../../../src/components/Tooltip';

const props = {
    placement: 'top', 
    iconClassName: 'icon icon-info', 
    className: '', 
    trigger: 'hover'
};
describe('ToastContainer Component', () => {
  it('renders correctly', () => {
    const component = shallow(<Tooltip {...props} />);
    expect(component).toMatchSnapshot();
  });
});
