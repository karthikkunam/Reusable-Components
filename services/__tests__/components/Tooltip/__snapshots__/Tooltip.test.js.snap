// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastContainer Component renders correctly 1`] = `
<div
  className=""
>
  <OverlayTrigger
    defaultShow={false}
    key="top"
    overlay={
      <Tooltip
        className="custom-tooltip bs-popover-top"
        id="tooltip-top"
        placement="right"
      />
    }
    placement="top"
    trigger="hover"
  >
    <button
      className="btn btn-sm btn-tooltip bs-tooltip"
      id="open-popover-link"
    >
      <span />
      <span
        className="icon icon-info"
      />
    </button>
  </OverlayTrigger>
</div>
`;
